### PR TITLE
Add review form

### PR DESCRIPTION
**Pull request by PBandJaimie**
@a-nouraldean @EJLEE808 @Taniel-P 

Hi friends! I have almost completed all form functionality. Probably the most difficult thing I've ever coded. I still need to figure out how to get photos to upload with thumbnail previews and then need to get a post method working on submit.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,3 @@ node_modules
 package-lock.json
 bundle.js
 coverage
-
-client/dist/bundle.js
-/Users/mini/Class/api/client/dist/bundle.js
-/client/dist/bundle.js
```
Comment (a-nouraldean) [inline .gitignore:9]: @PBandJaimie I think we need these files in gitignore?

File: client/dist/index.html [hunk 2]
```diff
@@ -5,7 +5,6 @@
     <link rel="stylesheet" type="text/css" href="styles.css">
     <link rel="stylesheet" type="text/css" href="./overviewCSS/overview.css">
     <link rel="stylesheet" type="text/css" href="./overviewCSS/OVmodal.css">
-    <link rel="stylesheet" type="text/css" href="./qnaCSS/qna.css">
```
Comment (a-nouraldean) [inline client/dist/index.html:8]: @PBandJaimie are you consolidating the CSS in one file?

File: server/server.js [hunk 3]
```diff
@@ -11,7 +11,6 @@ const relatedHelper = require('./routes/relatedProductsHelper.js');
 const reviews = require('./routes/reviewsRoutes');
 const reviewsHelper = require('./routes/reviewsHelpers.js');
 const qna = require('./routes/qnaRoutes')
-const qnaHelper = require('./routes/qnaHelper')
```
Comment (a-nouraldean) [inline server/server.js:14]: @PBandJaimie Is there a reason to delete this route?

File: server/server.js [hunk 4]
```diff
@@ -32,8 +31,7 @@ app.get('/getAllProductInfo', (req, res) => {
     relatedHelper.getProductID(req.query.id),
     reviewsHelper.getHelpfulReviews(req.query.id),
     reviewsHelper.getNewestReviews(req.query.id),
-    reviewsHelper.getMetaData(req.query.id),
-    qnaHelper.getQuestions(req.query.id)
```
Comment (a-nouraldean) [inline server/server.js:36]: @PBandJaimie Is there a reason to delete line 36?

File: server/routes/qnaRoutes.js [hunk 5]
```diff
@@ -1,32 +1,26 @@
...
-    const response = await axios.post(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rpp/qa/questions?product_id=${req.body.id}`,
-      {headers: {Authorization: config.TOKEN}});
-    res.sendStatus(201);
-  } catch (error) {
-    console.log('Error in post /questions: ', error);
-    res.sendStatus(404);
-  }
-}));
-
-module.exports = router;
```
Comment (a-nouraldean) [inline server/routes/qnaRoutes.js:31]: Please do not delete this route

File: client/src/app.jsx [hunk 6]
```diff
@@ -16,18 +16,18 @@ class App extends React.Component {
...
+      mainProductID: '36304',
       relatedProductIDs: '',
       characteristics: {},
       ratings: {},
       recommended: {},
       averageRating: 0,
       starRating: 0,
       helpfulReviews: [],
-      newestReviews: [],
-      questions: []
```
Comment (a-nouraldean) [inline client/src/app.jsx:30]: It looks like maybe you did not accept incoming changes on your merge request.

File: client/src/app.jsx [hunk 7]
```diff
@@ -130,12 +130,8 @@ class App extends React.Component {
           onClick={this.getAllProductInfo}
         />
         <br></br>
-        <QnA
-          renderStars={this.renderStars}
-          productId={this.state.mainProductID}
-          qData={this.state.questions}
-        />
-        <br></br>
```
Comment (a-nouraldean) [inline client/src/app.jsx:138]: This removes my change from the previous merge

File: client/src/components/overview/overview.jsx [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 import NewStyles from './components/newStyles.jsx'
-import NewPhotos from './components/newPhotos.jsx'
```
Comment (a-nouraldean) [inline client/src/components/overview/overview.jsx:3]: This is a previous change from @EJLEE808. It do not think it should be deleted.

File: client/src/app.jsx [hunk 9]
```diff
@@ -16,18 +16,18 @@ class App extends React.Component {
     this.state = {
       // cart: 'GET /cart
       overview: '',
+      name: '',
       styles: '',
       related: '',
-      mainProductID: '36300',
```
Comment (a-nouraldean) [inline client/src/app.jsx:21]: Good thing to change the current product id

File: client/src/app.jsx [hunk 10]
```diff
@@ -73,8 +74,7 @@ class App extends React.Component {
           starRating: data[6].avgRating.ratingPercentage,
           characteristics: data[6].characteristics,
           ratings: data[6].ratings,
-          recommended: data[6].recommended,
-          questions: data[7].results
```
Comment (a-nouraldean) [inline client/src/app.jsx:77]: This removes my change from the previous merge

File: client/src/components/QandA/QandA.jsx [hunk 11]
```diff
@@ -1,22 +1,18 @@
-/* eslint-disable react/prop-types */
 import React from 'react';
 import axios from 'axios';
 import SearchQuestion from './SearchQuestion.jsx';
 import TwoButtons from './TwoButtons.jsx';
-import QuestionList from './QuestionList.jsx'
```
Comment (a-nouraldean) [inline client/src/components/QandA/QandA.jsx:6]: This removes my change from the previous merge

File: client/src/components/QandA/QandA.jsx [hunk 12]
```diff
@@ -32,33 +28,22 @@ class QnA extends React.Component {
       question: this.state.questionSearch
     }
 
-    alert('Question search is currently disabled')
```
Comment (a-nouraldean) [inline client/src/components/QandA/QandA.jsx:35]: This removes my change from the previous merge

File: client/src/components/QandA/QuestionList.jsx [hunk 13]
```diff
@@ -1,21 +0,0 @@
...
-      <div className="questionList">
-        {this.props.qAndA.map((inquiry, index) =>
-          <Question key={index} question={inquiry.question_body} answer={inquiry.answers}/>
-        )}
-      </div>
-    )
-  }
-}
-
-export default QuestionList;
```
Comment (a-nouraldean) [inline client/src/components/QandA/QuestionList.jsx:21]: This removes my change from the previous merge

File: client/src/components/QandA/QandA.jsx [hunk 14]
```diff
@@ -32,33 +28,22 @@ class QnA extends React.Component {
...
 
   render () {
     return (
-      <div className='qnAWrapper'>
-        <header className='qHeader'><h3>QUESTIONS & ANSWERS</h3></header>
+      <div className='QnA'>
+        <div id='QStart'>QUESTIONS & ANSWERS</div>
         <SearchQuestion submit={this.onSubmit} cQuestion={this.changeQuestion} qsearch={this.state.questionSearch}/>
-        <QuestionList qAndA ={this.props.qData}/>
-        <TwoButtons loadQ={this.loadQuestions} addQ={this.addQuestion} />
```
Comment (a-nouraldean) [inline client/src/components/QandA/QandA.jsx:61]: This removes my change from the previous merge

File: client/src/components/QandA/SearchQuestion.jsx [hunk 15]
```diff
@@ -2,15 +2,15 @@ import React from 'react';
...
+    <div className='form-div'>
       <form onSubmit={props.submit}>
         <input type = 'text'
           placeholder='HAVE A QUESTION? SEARCH FOR ANSWERS...'
           onChange={props.cQuestion}
           value={props.qSearch}
-          className='form-control-form-group'
+          className='form-control form-group'
         />
-        <input type='submit' className='btn-btn-danger-btn-block' value='Submit'></input>
```
Comment (a-nouraldean) [inline client/src/components/QandA/SearchQuestion.jsx:13]: This removes my change from the previous merge

File: client/src/components/QandA/SearchQuestion.jsx [hunk 16]
```diff
@@ -2,15 +2,15 @@ import React from 'react';
...
 const SearchQuestion = (props) => {
   return (
-    <div className='qSearch'>
+    <div className='form-div'>
       <form onSubmit={props.submit}>
         <input type = 'text'
           placeholder='HAVE A QUESTION? SEARCH FOR ANSWERS...'
           onChange={props.cQuestion}
           value={props.qSearch}
-          className='form-control-form-group'
```
Comment (a-nouraldean) [inline client/src/components/QandA/SearchQuestion.jsx:11]: This removes my change from the previous merge

File: client/src/components/QandA/TwoButtons.jsx [hunk 17]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
 
-const TwoButtons = (props) => {
+const TwoButtons = () => {
   return (
-    <div className="twobuttons">
-      <button id='More-Answered-Questions' onClick={props.loadQ}>MORE ANSWERED QUESTIONS</button>
-      <button id='Add-Question' onClick={props.addQ}>ADD A QUESTION +</button>
```
Comment (a-nouraldean) [inline client/src/components/QandA/TwoButtons.jsx:7]: This removes my change from the previous merge

File: server/routes/qnaRoutes.js [hunk 18]
```diff
@@ -1,32 +1,26 @@
 const express = require('express');
 const router = express();
 const axios = require('axios');
+// const config = require('/home/bofowzi/Documents/rpp29/fec/api/config.js');
 const config = require('../../config.js');
 
-const wrap = fn => (...args) => fn(...args).catch(args[2]);
```
Comment (a-nouraldean) [inline server/routes/qnaRoutes.js:6]: This removes my change from the previous merge

File: client/src/spec/qna.spec.js [hunk 19]
```diff
@@ -2,50 +2,21 @@
...
-
-    expect(wrapper.find('.questionList')).to.have.lengthOf(1);
-  });
-
-});
+// eslint-disable-next-line func-style
+function sayHello (string) {
+  let response = 'Hello ' + string + '!';
+  return response;
+}
```
Comment (a-nouraldean) [inline client/src/spec/qna.spec.js:19]: This removes my change from a previous merge

File: client/src/components/Reviews/AddReview.jsx [hunk 20]
```diff
@@ -0,0 +1,18 @@
+import React, {useState} from 'react';
```
Comment (a-nouraldean) [inline client/src/components/Reviews/AddReview.jsx:1]: Good state management here

File: client/src/components/Reviews/FactorBar.jsx [hunk 21]
```diff
@@ -26,20 +26,20 @@ const FactorBar = (props) => {
...
+    rightDescriptor = 'Runs long';
   }
 
   if (props.name === 'Fit') {
-    leftDescriptor = 'Tight';
+    leftDescriptor = 'Runs tight';
     centerDescriptor = 'Perfect';
-    rightDescriptor = 'Loose';
+    rightDescriptor = 'Runs loose';
   }
```
Comment (a-nouraldean) [inline client/src/components/Reviews/FactorBar.jsx:44]: Good job getting this done.

File: client/src/components/Reviews/FormCharacteristic.jsx [hunk 22]
```diff
@@ -0,0 +1,148 @@
+import React from 'react';
```
Comment (a-nouraldean) [inline client/src/components/Reviews/FormCharacteristic.jsx:1]: Good job getting this done!

File: client/src/components/Reviews/ReviewForm.jsx [hunk 23]
```diff
@@ -1,8 +1,269 @@
...
+    if (this.state.Length > 0) {
+      charTotal += 1;
+    }
+    if (this.state.Fir > 0) {
+      charTotal += 1;
+    }
+    return charTotal;
+  }
+
+  handleStarClick(event) {
```
Comment (a-nouraldean) [inline client/src/components/Reviews/ReviewForm.jsx:82]: Good job!

**Review by a-nouraldean — CHANGES_REQUESTED** [review]
Hi @PBandJaimie! Your code looks good. 

There are places that it seems you did not accept incoming changes from a merge. I am around if you want to discuss.

**Comment by PBandJaimie** [comment]
> Hi @PBandJaimie! Your code looks good.
> 
> There are places that it seems you did not accept incoming changes from a merge. I have highlighted some of the items that were deleted. Let's review after the standup.

Hey! I’m so sorry! I don’t know what happened with my PR. I swear i merged that code and didn’t delete all your pieces! I gotta figure it out.

**Pull request closed without merging**